src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsTaggerProvider.cs

```diff
@@ -121,5 +121,19 @@ protected internal override bool IncludeDiagnostic(DiagnosticData diagnostic)
                 return null;
             }
         }
+
+        /// <summary>
+        /// TODO: is there anything we can do better here? Inline diagnostic tags are not really data, but more UI
```

Comment:
@akhera99?

Reply:
I think what the comment says makes sense, separating the data tagging and creating UI tags from the data tags. I would be happy to change that in a follow-up PR.